### ISSUE TITLE
Single Network in adapter_wd & adapter_dcn

### ISSUE THREAD

**Issue by idoblankfiverr** [Applied-Machine-Learning-Lab/HAMUR#4]
Hi,
I was wondering why in the adapter_wd and adapter_dcn you initialize only one wd/dcn network, and not one per domain (as you do in the adapter mlp section)?
If you use the same network for all the domains, how is the multi domain take into account?

From adapter_dcn - you are looping over all the domains, but having the same output again and again (only the masking is different)

        for d in range(self.num_domains):
            domain_mask = (domain_id == d)
            mask.append(domain_mask)
            domain_input = embed_x
            cn_out = self.cn(domain_input)
            mlp_out = self.mlp(domain_input)
            x_stack = torch.cat([cn_out, mlp_out], dim=1)
            y = self.linear(x_stack)
            out.append(torch.sigmoid(y))

Thanks for your advice!

**Comment by Xiaopengli1** [comment]
Thank you for your question, @idoblankfiverr .

Regarding the first question about the implementation of DCN_adp and WD_adp, during the implementation of different backbone experiments, our target baselines are STAR and DCN, W&D directly used in MDR. Therefore, we built different MLPs for different domains but solely used DCN and W&D for MDR. I thinks it is worth trying building independent DCN or W&D models for different domains. However, you should consider the efficiency and parameter size for a fair comparison with HAMUR and STAR.

For the second question, it is actually a trick we implemented in the MDR code. Two adjacent instances in one batch may come from different domains, and it is not easy to recover to the original order if domain data streamings are gathered and calculated independently. Hence, we use different domain-IDs as masks for three cycles, use embed_x to store original batch data. This ensures that domains are calculated separately while maintaining the sequence of instances in the data flow.

Thank you again for your attention to our work.
